storm-core/src/clj/backtype/storm/cluster.clj

```diff
@@ -518,6 +540,13 @@
           (catch KeeperException e
             (log-warn-error e "Could not teardown errors for " storm-id))))
 
+      (teardown-topology-log-config!
```

Comment:
I saw that you are calling it from cleanup thread,  I am just wondering why not call cleanup with all the other topology cleanup methods? 
From what I understand the nimbus cleanup thread is designed to identify topologies that have no nimbus local disk state but their zookeeper state is somehow not cleared up. It seems like a handler for rare failure scenarios which we are abusing now. Let me know if you have some other reason for not deleting this with rest of the topology.